File: Increment.java
//Increment.java
//Prefix increment and postfix increment operators.

public class ClassAverage
{
   public static void main(String[] args)
   {   
      // demonstrate postfix increment operator
      int c = 7; 
      System.out.printf("c before postincrement: %d%n", c); // prints 7
      System.out.printf("    postincrementing c: %d%n", c++); // prints 7
      System.out.printf(" c after postincrement: %d%n", c); // prints 8

      System.out.println(); // skip a line

      // demonstrate prefix increment operator
      c = 5; 
      System.out.printf(" c before preincrement: %d%n", c); // prints 7
      System.out.printf("     preincrementing c: %d%n", ++c); // prints 8
      System.out.printf("  c after preincrement: %d%n", c); // prints 8
   } 
} // end class 

File: Increment.cpp
//Increment.cpp
//Prefix increment and postfix increment operators.
#include <iostream>
using namespace std;

int main()
{
		int c = 7;
		cout<<"c before postincrement: "<< c<<endl; // prints 7
		cout << "    postincrementing c: "<< c++ << endl; // prints 7
		cout << " c after postincrement: "<< c << endl; // prints 8

		cout << "\n"; // skip a line

		// demonstrate prefix increment operator
		c = 7;
		cout << " c before preincrement: "<< c << endl; // prints 7
		cout << "     preincrementing c: "<< ++c << endl; // prints 8
		cout << "  c after preincrement: "<< c << endl; // prints 8
		system("pause");
} // end class 
